File: JAVA_ARRAYS/Sort012.java
package JAVA_ARRAYS;

import java.util.Arrays;

public class Sort012 {
    public static void main(String[] args) {
        int arr[]={0,2,1,2,2,0,1};
        Sort012(arr);
        System.out.println(Arrays.toString(arr));
    
    }
    public static void Sort012(int[]a){
        int n=a.length;
        int l=0,m=0,h=n-1;
        while(m<=h){
            if(a[m]==0){
                Swap(a, m, l);
                l++;
                m++;  
            }
            else if(a[m]==1){
                m++;
            }
            else{
                Swap(a, m, h);
                h--;
            }
        }
    }
    public static void Swap(int []arr,int a,int b){
        int temp= arr[a];
        arr[a]=arr[b];
        arr[b]=temp;
    }
    
}

File: Solved ques/sort012.cpp
#include<bits/stdc++.h>
using namespace std;

void sort012(vector<int>&arr,int n){
    int L=0,M=0,H=n-1;

    while(M<=H){
        if(arr[M]==0){
            swap(arr[L],arr[M]);
            L++;
            M++;

        }
        else if(arr[M]==1){
                M++;
            
        }
        else{
            swap(arr[M],arr[H]);
            H--;

        }
     
    }
}

int main(){
    vector<int> arr={0,1,2,0,1,2,0,1,2};
    int n=arr.size();
    sort012(arr,n);

    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    

}